### ISSUE TITLE
Wrk/Wrk2 duration won't match the original cmd

### ISSUE THREAD

**Issue by franz1981** [Hyperfoil/Hyperfoil#268]
Wrk/Wrk2 test duration is now adding a calibration phase that makes tests to last longer then the original commands they should mimic.

**Comment by johnaohara** [comment]
What is the difference in time being reported vs time defined in the cl args?  

there is a calibration phase that is performed before the test, but that phase should not be reported in the wrk/wrk2 stats, only the measurements in the `test` phase are reported. 

There might be a slight discrepancy in that the duration is calculated from the `endTime` - `startTime` for the test phase, which may overrun if waiting for results to be returned at the end of the test phase

**Comment by franz1981** [comment]
Adding `time` prefix for a `wrk -t 1 -c 10 -d 10` etc etc run reports:
```
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   102.97μs   40.82μs   2.23ms   79.04%
    Req/Sec   80729.00  25608.78  91294.00     90.91
  888019 requests in 10.001s, 124.49MB read
Requests/sec: 88793.02
Transfer/sec:  12.45MB

real    0m19.214s
user    0m16.058s
sys     0m6.569s
```
The additional seconds seems too much to me, given the latency profile, and indeed hyperfoil CLI shows:
![image](https://user-images.githubusercontent.com/13125299/214298144-be1d0f5e-a8f7-4019-a53a-7d74626d1c6e.png)
that's 6 seconds duration of calibration and 9 seconds of test

**Comment by johnaohara** [comment]
The calibration phase is deterministic, i.e. is the same no matter input params.  We need to consider when a large number of connections and users (threads) have been defined, and the impact the config has on warming up the load driver

**Comment by johnaohara** [comment]
wrk is written in c, hyperfoil in java, we need to JIT to compile the driver code and allocate the memory so that JIT and GC do not impact the results. That is the purpose of the calibration phase. Without it, we would not have comparable results. 

the output is the same as wrk, albeit that the duration is reported as 10ms longer, but that is well within the rounding error that wrk reports, i.e. `10.001s` vs `10.00s`

**Comment by franz1981** [comment]
> we need to JIT to compile the driver code and allocate the memory so that JIT and GC do not impact the results

yeah, but that's an implementation detail, having a command which requires 10 seconds, to use 50% more time, is surprising, although reasonable. 
Probably it worths to ship `wrk/wrk2` as native built commands - I won't expect any calibration to be required there

**Comment by johnaohara** [comment]
The calibration phase is fixed, the `%` impact of total runtime depends on the duration of the test. 

wrk/wrk2 will not compile to native as-is. They are just predefined benchmarks that are deployed to a local hyperfoil instance. 

The output of the command was intended to be a drop-in replacement for wrk/wrk2, (i.e. the output should be the same as well for parsing reasons).  We can (probably should) cover in documentation. 

I understand that if you are relying on a process taking a set amount of time, then there can/will be differences, I'm not sure that waiting a fixed length of time for a process to complete is a robust/repeatable strategy, we might want to point that out
